include/coap2/coap_debug.h

```diff
@@ -50,6 +50,14 @@ typedef enum {
 } coap_log_t;
 #endif
 
+#ifndef LOG_CIPHERS
+/*
+   LOG_DEBUG+2 gives ciphers in GnuTLS
+   Use LOG_CIPHERS to output Cipher Info in OpenSSL etc.
+ */
+#define LOG_CIPHERS LOG_DEBUG+2
```

Comment:
At this point we should think about extending `coap_log_t` as this construct is weird.
The problem, of course, is compatibility (and consistency) with syslog levels. In the long run, we will need to prefix the log levels, resulting in something like this:

```c
typedef enum {
  COAP_LOG_EMERG=LOG_EMERG,
  ...
  COAP_LOG_DEBUG,
  COAP_LOG_CIPHERS=LOG_CIPHERS
} coap_log_t;
```

Reply:
I was tempted to update coap_log_t, but there are 2 definitions for it with the Linux one being just a short as I did like my solution that much.  Changing all the coap_log() first parameters to COAP_ prefix is a major change.  As you say, long term this is something that should be properly sorted out.
In my code usage, I do use coap_set_log_handler() which points to a function that uses syslog(), so ideally the one-to-one mapping needs to remain.

Reply:
Yes, the mapping should clearly be kept. I just ran into issues with RIOT-OS where only some of the syslog levels have been defined.
I am willing to do the change at some point but would start with a minor change that allows keeping the old values for now. I can make a proposal for this but first we may want to answer the question where LOG_CIPHERS lives.

Reply:
I will come up with something later today for this.

Reply:
An alternative for handling LOG_CIPHERS with some re-organization has been pushed.

Things are complicated in that client.c does not have HAVE_SYSLOG_H defined and it is not easy to auto add a COAP_ prefix to the coap_log() first parameter as this parameter can either be a LOG_ #define/enum or an actual numerical coap_log_t value.

Reply:
I am not sure I understand your concern. But the proposed solution is good enough for this PR, so we should do the houskeeping later in a different PR.